compiler/passes/normalize.cpp

```diff
@@ -1714,9 +1714,10 @@ static void change_method_into_constructor(FnSymbol* fn) {
     INT_FATAL(fn, "'this' argument has unknown type");
 
   // Now check that the function name matches the name of the type
-  // attached to 'this'.
-  if (strcmp(fn->getFormal(2)->type->symbol->name, fn->name) &&
-      strcmp(fn->name, "init"))
+  // attached to 'this' or matches 'init'.
+  bool notCtor = strcmp(fn->getFormal(2)->type->symbol->name, fn->name);
+  bool notInit = strcmp(fn->name, "init");
```

Comment:
I think this code would be much clearer if you had variables "isCtor" and "isInit".
I'd set these as e.g. bool isInit = (0== strcmp(...) );